config/cadical.m4

```diff
@@ -0,0 +1,91 @@
+# CVC4_CHECK_FOR_CADICAL
+# ------------------
+# Look for CaDiCaL and link it in, but only if user requested.
+AC_DEFUN([CVC4_CHECK_FOR_CADICAL], [
+AC_MSG_CHECKING([whether user requested CaDiCaL support])
+
+have_libcadical=0
+CADICAL_LIBS=
+CADICAL_LDFLAGS=
+
+have_libcadical=0
+if test "$with_cadical" = no; then
+  AC_MSG_RESULT([no, CaDiCaL disabled by user])
+elif test -n "$with_cadical"; then
+  AC_MSG_RESULT([yes, CaDiCaL requested by user])
+  AC_ARG_VAR(CADICAL_HOME, [path to top level of CaDiCaL source tree])
+  AC_ARG_WITH(
+    [cadical-dir],
+    AS_HELP_STRING(
+      [--with-cadical-dir=PATH],
+      [path to top level of CaDiCaL source tree]
+    ),
+    CADICAL_HOME="$withval",
+    [ if test -z "$CADICAL_HOME" && ! test -e "$ac_abs_confdir/cadical/build/libcadical.a"; then
+        AC_MSG_FAILURE([must give --with-cadical-dir=PATH, define environment variable CADICAL_HOME, or use contrib/get-cadical to setup CaDiCaL for CVC4!])
+      fi
+    ]
+  )
+
+  # Check if CaDiCaL was installed via contrib/get-cadical
+  AC_MSG_CHECKING([whether CaDiCaL was already installed via contrib/get-cadical])
+  if test -z "$CADICAL_HOME" && test -e "$ac_abs_confdir/cadical/build/libcadical.a"; then
+    CADICAL_HOME="$ac_abs_confdir/cadical"
+    AC_MSG_RESULT([yes, $CADICAL_HOME])
+  else
+    AC_MSG_RESULT([no])
+  fi
+
+  if ! test -d "$CADICAL_HOME" || ! test -e "$CADICAL_HOME/build/libcadical.a" ; then
+    AC_MSG_FAILURE([either $CADICAL_HOME is not a CaDiCaL source tree or it's not yet built $CADICAL_HOME/build/libcadical.a])
+  fi
+
+  CPPFLAGS="$CPPFLAGS -I$CADICAL_HOME/src"
```

Comment:
It looks like this should be together with lines 76/77. You are adding the folder as an include path in configure.ac, so no need to modify the `CPPFLAGS` globally.

Reply:
No,  76/77 is only set temporarily for the `AC_LINK_IFELSE` check. After the check the flags get restored.